cfpb/consumerfinance.gov
# Stub Mapbox API in Cypress tests

Similar to #7271 and #7277, this change stubs Mapbox API responses in Cypress tests. This follows the trend set in those previous PRs to stub responses from external APIs so that our functional tests don't fail if an external service is slow or unreachable. This will also resolve an issue we're having with these tests failing in our EKS environment.

---

## Additions

- Mapbox API fixtures

## Changes

- Intercept and stub Mapbox APIs in Cypress tests

## How to test this PR

1. Run the Find a Housing Counselor Cypress tests with `yarn cypress open` and selecting the `find-a-housing-counselor.cy.js` spec
2. Note the `(xhr) GET 200 https://api.mapbox.com/styles/v1/mapbox/streets-v11?access_token=None` and `(xhr) GET 200 https://api.mapbox.com/v4/mapbox.mapbox-streets-v8,mapbox.mapbox-terrain-v2.json?access_token=None&secure` in step 7 show that the responses were stubbed (see screenshot below)

## Screenshots

Before | After
--- | ---
![before](https://user-images.githubusercontent.com/1862695/194641703-7ee2817c-d5f3-4a40-bcaf-471ec1f84882.png) | ![after](https://user-images.githubusercontent.com/1862695/194641727-fb951843-3e25-4683-8bd0-90e8802fff3e.png)

## Notes and todos

- I tried to make the intercepted URLs a little flexible by sticking a `\d+` in for the version number. On one hand, that will keep intercepting the requests if the API version number changes. On the other hand, that might mask any issues our tool has with a new Mapbox API version. In general we've been trying to decouple our functional tests from external APIs, though, so this seemed like the right tradeoff (especially since our functional tests are now required to pass for PRs and automated deployments).
- Relatedly, if the responses from Mapbox APIs ever do change, we'll have to remember to update the fixtures.
- If you don't have a Mapbox access token set, you'll still get a bunch of `401` errors when the page tries to request the map tiles, but since those aren't required for tests to pass, I'm fine leaving those.

## Checklist

  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
